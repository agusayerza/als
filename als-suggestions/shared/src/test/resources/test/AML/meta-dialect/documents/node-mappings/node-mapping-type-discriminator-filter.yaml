#%Dialect 1.0
dialect: Test Unions
version: 1.0

nodeMappings:
  A:
    classTerm: vocab.thing
    mapping:
      text:
        propertyTerm: vocab.text
        range: string
  B:
    classTerm: vocab.something
    mapping:
      text:
        propertyTerm: vocab.text
        range: string

  RootNode:
    union:
      - A
      - B
    typeDiscriminatorName: kind
    typeDiscriminator:
      TypeB: B
      TypeA: *