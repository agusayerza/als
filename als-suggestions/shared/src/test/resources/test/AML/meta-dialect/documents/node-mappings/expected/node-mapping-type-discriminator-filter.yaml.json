[
  {
    "label": "A",
    "kind": [],
    "detail": "unknown",
    "documentation": "A",
    "deprecated": [],
    "preselect": [],
    "sortText": "11A",
    "filterText": "A",
    "insertText": null,
    "insertTextFormat": [
      1
    ],
    "textEdit": {
      "range": {
        "start": {
          "line": 25,
          "character": 13
        },
        "end": {
          "line": 25,
          "character": 13
        }
      },
      "newText": "A"
    },
    "additionalTextEdits": null,
    "commitCharacters": null,
    "command": null
  }
]